street2sat_utils/client.py

```diff
@@ -1,32 +1,49 @@
-from __future__ import annotations
-
 import base64
 import io
 import math
-from collections import Counter, defaultdict
+from collections import defaultdict
 from dataclasses import dataclass
 from datetime import datetime
 from statistics import mean
-from typing import Dict, IO, List, Optional, Tuple, Union
+from typing import IO, Dict, List, Optional, Tuple
 
 import cv2  # type: ignore
 import exifread  # type: ignore
 import numpy as np
 from matplotlib.figure import Figure  # type: ignore
 from yolov5 import hubconf  # type: ignore
-from yolov5.models.yolo import Model  # type: ignore
 
-from constants import CROP_CLASSES, CROP_TO_HEIGHT_DICT, GOPRO_SENSOR_HEIGHT, MODEL_PATH
 from street2sat_utils import exif_utils
+from street2sat_utils.constants import (
+    CROP_CLASSES,
+    CROP_TO_HEIGHT_DICT,
+    GOPRO_SENSOR_HEIGHT,
+    MODEL_PATH,
+)
+
+
+def memoize(f):
+    memo = {}
+
+    def helper(x):
+        if x not in memo:
+            memo[x] = f(x)
+        return memo[x]
+
+    return helper
 
-model: Model = hubconf.custom(str(MODEL_PATH))
-model.eval()
+
+@memoize
+def load_model(model_path: str):
+    model = hubconf.custom(model_path)
+    model.eval()
+    return model
```

Comment:
Added a memoized function for loading the model so it's only loaded when needed and only loaded once.